src/ConnectionPool.jl

```diff
@@ -686,27 +684,34 @@ function tcpstatus(c::Connection)
 end
 
 function showpool(io::IO)
-    lock(poollock)
-    println(io, "ConnectionPool[")
-    for c in pool
-        println(io, "   $c")
+    lock(poollock[Threads.threadid()]) do
+        println(io, "ConnectionPool[")
+        for c in pool[Threads.threadid()]
+            println(io, "   $c")
+        end
+        println(io, "]\n")
     end
-    println(io, "]\n")
-    unlock(poollock)
 end
 
 function showpoolhtml(io::IO)
-    lock(poollock)
-    println(io, "<table>")
-    for c in pool
-        print(io, "<tr>")
-        for x in split("$c")
-            print(io, "<td>$x</td>")
+    lock(poollock[Threads.threadid()]) do
+        println(io, "<table>")
+        for c in pool[Threads.threadid()]
+            print(io, "<tr>")
+            for x in split("$c")
+                print(io, "<td>$x</td>")
+            end
+            println(io, "<tr>")
         end
-        println(io, "<tr>")
+        println(io, "</table>")
     end
-    println(io, "</table>")
-    unlock(poollock)
+end
+
+function __init__()
+    Threads.resize_nthreads!(pool)
+    Threads.resize_nthreads!(poollock)
+    Threads.resize_nthreads!(poolcondition)
```

Comment:
These arrays are invalid to pass to `resize_nthreads!`

Reply:
It would be infinitely more helpful to explain why or to provide an alternative solution. Nothing from the `resize_nthreads!` docs seems to indicate that anything would be wrong with this approach.